app/services/create_work_package_service.rb

```diff
@@ -0,0 +1,52 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+class CreateWorkPackageService
+  attr_accessor :user, :project, :work_package
+
+  def initialize(user:, project:, send_notifications: true)
+    self.user = user
+    self.project = project
+
+    WorkPackageObserver.instance.send_notification = send_notifications
+  end
+
+  def create
+    hash = {
+      project: project,
+      author: user,
+      type: project.types.where(is_default: true).first || project.types.first
+    }
+    self.work_package = project.add_work_package(hash)
+  end
+
+  def save
+    work_package.save
```

Comment:
As @ulferts already hinted, the proposed solution would require quite a refactoring. Since this PR is already pretty large I wrote up a work package to be done in the future: https://community.openproject.org/work_packages/20322
For this PR I will then implement the first proposal of @NobodysNightmare.